src/material.cpp

```diff
@@ -243,8 +243,13 @@ void pyne::Material::from_hdf5(std::string filename, std::string datapath, int r
     H5O_info_t object_info;
     status = H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
     status = H5Oget_info_by_name(db, "/material" , &object_info, H5P_DEFAULT);
+    herr_t status2;
```

Comment:
Can we add something like this to help document what we are doing?  i.e. I don't actually know that values of `status` or the logic, but this will make it easier.
```suggestion
    herr_t status2;
    bool material_path_exists = (status ==0 && object_info.type != H5O_TYPE_DATASET)
```
We can then use it in the `if` statement in line 251

Reply:
Similar for status2....